runtime/util_core/j9argscan.c

```diff
@@ -111,8 +115,25 @@ void scan_failed(J9PortLibrary * portLibrary, const char* module, const char *sc
 uintptr_t scan_double(char **scan_start, double *result)
 {
 	char *endPtr = NULL;
-
+#if defined(WINDOWS)
+	_locale_t clocale = _create_locale(LC_ALL, "C");
+	*result = _strtod_l(*scan_start, &endPtr, clocale);
+	_free_locale(clocale);
+#elif defined(OSX) /* defined(WINDOWS) */
+	locale_t clocale = newlocale(LC_ALL_MASK, "C", NULL);
+	*result = strtod_l(*scan_start, &endPtr, clocale);
+	freelocale(clocale);
+#elif defined(LINUX) || defined(LC_ALL_MASK) /* defined(WINDOWS) */
+	/* AIX 7.1+ has support for newlocale() */
+	locale_t clocale = newlocale(LC_ALL_MASK, "C", NULL);
+	locale_t orglocale = uselocale(clocale);
+	*result = strtod(*scan_start, &endPtr);
+	uselocale(orglocale);
+	freelocale(clocale);
+#else  /* defined(WINDOWS) */
+	/* No support on AIX 6.1 or z/OS so far, but testing shows strtod() isn't locale dependent */
 	*result = strtod(*scan_start, &endPtr);
+#endif /* defined(WINDOWS) */
```

Comment:
Each of `_create_locale()` and `newlocale()` can fail, returning `NULL`. That should be checked, in which case, we could revert to using `strtod()`.

nit: extra space after `#else`

Reply:
Updated.